packages/user_packages/iotcc_mqtt.py

```diff
@@ -76,50 +99,73 @@ def run(self, registry):
+            # Trying to establish MQTT Connection with retry attempts in case of exception
+            while conn_attempts <= retry_attempts:
+                try:
+                    self.iotcc = IotControlCenter(
+                        MqttDccComms(edge_system_name=edge_system.name, url=registry.get("broker_ip"),
+                                     port=registry.get("broker_port"), identity=identity, tls_conf=tls_conf,
+                                     client_id=registry.get("broker_username"), enable_authentication=True,
+                                     mqtt_msg_attr=mqtt_msg_attr))
+                    break
+                except Exception, e:
+                    if conn_attempts == retry_attempts:
+                        raise
+                    conn_attempts += 1
+                    log.error('MQTT Connection failed - {0}'.format(str(e)))
+                    log.info('Trying MQTT Connection: Attempt - {0}'.format(str(conn_attempts)))
+                    time.sleep(mqtt_connection_delay_retries)
+
+            # Attempts for Edge System Registration
+            reg_attempts = 0
+            # Edge System Registration with retry attempts in case of exception
+            while reg_attempts <= retry_attempts:
                 try:
-                    # Register edge system (gateway)
-                    self.iotcc.set_properties(self.iotcc_edge_system, {"key1": "value1", "key2": "value2"})
+                    self.iotcc_edge_system = self.iotcc.register(edge_system)
                     break
-                except Exception:
-                    # In the third attempt if get exception raise it
-                    if attempts == max_retry_attempts:
+                except Exception, e:
+                    if reg_attempts == retry_attempts:
                         raise
-                    attempts += 1
-                    # The sleep time before re-trying depends on the infrastructure requirement of broker to restart
-                    # It can be modified or removed as per the infrastructure requirement
-                    time.sleep(5)
+                    reg_attempts += 1
+                    log.error('MQTT Connection Failed - {0}'.format(str(e)))
+                    log.info('Trying Edge System {0} Registration: Attempt - {1}'.format(edge_system.name,
+                                                                                         str(reg_attempts)))
+                    time.sleep(delay_retries)
+
             registry.register("iotcc_mqtt", self.iotcc)
-            # Store the registered edge system object in liota package manager registry after the
             registry.register("iotcc_mqtt_edge_system", self.iotcc_edge_system)
 
-        except RegistrationFailure:
-            print "EdgeSystem registration to IOTCC failed"
+            # Attempts for setting edge system properties
+            prop_attempts = 0
+            # Set Properties with retry attempts in case of exception
```

Comment:
Setting Multiple Properties of Gateway by passing Dictionary Object.

Reply:
Added